core/src/test/scala/unit/kafka/server/HeartbeatRequestTest.scala

```diff
@@ -146,37 +145,33 @@ class HeartbeatRequestTest(cluster: ClusterInstance) extends GroupCoordinatorBas
         metadata = metadata
       )
 
-      val reJoinFollowerFuture = Future {
+      Future {
```

Comment:
Do we need this one as well?

Reply:
This is necessary because sendJoinRequest gets blocked if not all members have rejoined

Reply:
Or we can also remove the future by removing the `waitUntilTrue` in sendJoinRequest

Reply:
gotcha. makes sense, thanks.

Reply:
Hmm, I tried to remove this but it doesn't work. Unless we return a future in sendJoinRequest so the future is needed anyway

Reply:
Yeah, it is because the server does not actually reply until all the members have rejoin the group.